src/components/spotlight/Spotlight.tsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import { PropsWithChildren } from 'react';
+import { useIntl } from 'react-intl';
+import ReactJoyride, { TooltipRenderProps } from 'react-joyride';
+import { SpotlightModalForStep } from './SpotlightModalForStep';
+import { SpotlightProps, SpotlightStep } from './SpotlightTypes';
+
+export function Spotlight(props: Readonly<SpotlightProps>) {
```

Comment:
Are we missing the forwardRef that would attach to SpotlightModalStep and then to the SpotlightModal?